anchor/models/post.php

```diff
@@ -59,6 +59,15 @@ public static function search($term, $page = 1, $per_page = 10) {
 				Base::table('users.bio as author_bio'),
 				Base::table('users.real_name as author_name')));
 
+		foreach ($posts as $key => $post) {
+			if ($post->data['status'] !== 'published') {
+				unset($posts[$key]);
+			}
+		}
+		if (count($posts) < 1) {
```

Comment:
This isn't my code - I've rebased the PR and resubmitted as #819.

> In [anchor/models/post.php](https://github.com/anchorcms/anchor- 
> cms/pull/813%23discussion_r24715405):
 > @@ -59,6 +59,15 @@ public static function search($term, $page = 1, $per_page = 10) { > Base::table('users.bio as author_bio'), > Base::table('users.real_name as author_name'))); > > + foreach ($posts as $key => $post) { > + if ($post->data['status'] !== 'published') { > + unset($posts[$key]); > + } > + } > + if (count($posts) < 1) { > Why not using $query->count()? And this is repeating the $total, maybe a 
> shorthanded if/else form here to replace the one above?
> — 
>  
> Reply to this email directly or [view it on 
> GitHub](https://github.com/anchorcms/anchor- 
> cms/pull/813/files%23r24715405).![](https://github.com/notifications/beacon/AEfHRd_9i5yj_e7ZUHLF_Xv4CqzCx8Obks5nr3HwgaJpZM4DaVHp.gif)